client.go

```diff
@@ -262,6 +273,24 @@ func (c *Client) updateAuth(reg *protocol.RegisterResult) {
 	}
 }
 
+func (c *Client) setContext() {
```

Comment:
should we remove this func and make `context()` idempotent?

Reply:
It's not possible, a single Client may have multiple sessions (due to reconnects). Each session must have new context.

Reply:
👍